utils/common/src/main/java/org/apache/brooklyn/util/repeat/Repeater.java

```diff
@@ -141,6 +148,27 @@ public Repeater repeat(Callable<?> body) {
     }
 
     /**
+     * Use a new thread for every iteration of the loop.
+     *
+     * @return {@literal this} to aid coding in a fluent style.
+     */
+    public Repeater threaded() {
+        this.executor = Executors.newSingleThreadExecutor();
+        this.shutdown = true;
+        return this;
+    }
+
+    /**
+     * @see #threaded()
+     * @param executor an externally managed {@link ExecutorService} to use when creating threads.
+     * @return {@literal this} to aid coding in a fluent style.
+     */
+    public Repeater threaded(ExecutorService executor) {
+        this.executor = executor;
+        return this;
```

Comment:
A bit on the pedantic side - can you add `shutdown =false` here. Can imagine first calling `threaded()` and passing the builder to another method only to add a custom executor using `threaded(myExecutor)`. Good to merge after that.